src/Microsoft.DotNet.Wpf/src/WindowsBase/System/Windows/UncommonField.cs

```diff
@@ -54,12 +53,24 @@ public void SetValue(DependencyObject instance, T value)
                 // Set the value if it's not the default, otherwise remove the value.
                 if (!EqualityComparer<T>.Default.Equals(value, _defaultValue))
                 {
-                    instance.SetEffectiveValue(entryIndex, null /* dp */, _globalIndex, null /* metadata */, value, BaseValueSourceInternal.Local);
+                    object valueObject;
+
+                    if (typeof(T) == typeof(bool))
```

Comment:
May I know the performance difference between `typeof(T) == typeof(bool)` and `T is bool`? Thank you.

Reply:
`typeof(T) == typeof(bool)` and `value is bool` should both be optimized to a const value by the JIT in a release build and tier 1 compilation.  In tier 0, it's possible `value is bool` could box.

Reply:
Thank you very much.